pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/actions/RestartDeclarativePipelineAction/index.jelly

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?jelly escape-by-default='true'?>
+<!--
+  ~ The MIT License
+  ~
+  ~ Copyright (c) 2017, CloudBees, Inc.
+  ~
+  ~ Permission is hereby granted, free of charge, to any person obtaining a copy
+  ~ of this software and associated documentation files (the "Software"), to deal
+  ~ in the Software without restriction, including without limitation the rights
+  ~ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  ~ copies of the Software, and to permit persons to whom the Software is
+  ~ furnished to do so, subject to the following conditions:
+  ~
+  ~ The above copyright notice and this permission notice shall be included in
+  ~ all copies or substantial portions of the Software.
+  ~
+  ~ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  ~ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  ~ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  ~ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  ~ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  ~ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  ~ THE SOFTWARE.
+  -->
+
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:l="/lib/layout" xmlns:f="/lib/form">
+    <l:layout title="Restart ${it.owner.displayName} from Stage" norefresh="true">
+        <st:include page="sidepanel.jelly" it="${it.owner}"/>
+        <l:main-panel>
+            <j:if test="${it.restartEnabled}">
+                <h1>Restart ${it.owner.displayName} from Stage</h1>
+                <p>
+                    <j:out value="Help text to go here"/>
+                </p>
+                <f:form action="restart" method="POST" name="restart">
+                    <f:entry field="stageName" title="Stage Name">
+                        <!-- TODO: Help! -->
+                        <f:textarea checkUrl="'${it.checkUrl}?value='+encodeURIComponent(this.value)"/>
```

Comment:
Do you not just want an `f:select` here? Or even a list of stages, each one with a **Restart from here** button?

Reply:
I'm basically tabling the UI aspects until I huddle with someone who actually knows UI. =)